File: 🟢 Easy/283 - Move Zeroes/_283MoveZeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        countZero = 0
        j = 0
        for i in range(0, len(nums)):
            if nums[i] == 0:
                countZero += 1
            else:
                nums[j] = nums[i]
                j += 1

        for i in range(0, countZero):
            nums[j] = 0
            j += 1

File: 🟢 Easy/283 - Move Zeroes/_283MoveZeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int countZero = 0;
        int j = 0;
        for (int i = 0; i < nums.size(); i++) {
            if (nums[i] == 0) {
                countZero++;
            } else {
                nums[j++] = nums[i];
            }
        }

        for (int i = 0; i < countZero; i++) {
            nums[j++] = 0;
        }
    }
};